src/components/Tool/Appeal/Modals/UpdateDonationsModal/DonationTable/DonationTable.tsx

```diff
@@ -94,21 +101,20 @@ export const DonationTable: React.FC<DonationTableProps> = ({
   });
 
   useEffect(() => {
-    if (!data?.donations.nodes.length) {
-      return;
+    if (preselectedDonations?.length) {
+      setSelectedDonations(preselectedDonations);
     }
-
-    const preselectedDonations = data.donations.nodes
-      .filter((donation) => donation.appeal?.id === appealId)
-      .map(createDonationRow);
-
-    setSelectedDonations(preselectedDonations);
-  }, [data]);
+  }, [preselectedDonations]);
 
   const nodes = data?.donations.nodes || [];
 
   const donations = useMemo(() => nodes.map(createDonationRow), [nodes]);
 
+  const isPreselectedDonation = (donation: DonationRow) =>
+    preselectedDonations.some((donationRow) => donationRow.id === donation.id);
```

Comment:
Use find instead of some so if it finds one it stops looping through the array.
```suggestion
    !!preselectedDonations.find((donationRow) => donationRow.id === donation.id);
```

Reply:
`some` and `every` short-circuit also. Run this in the browser console and see that it only logs once.

```js
[1, 2, 3].some(() => {
  console.log('callback');
  return true;
});
```

I think that `some` is more appropriate semantic choice here because we're trying to check that an element exists in the array and don't need to know what the actual element is.

Reply:
I'm sorry, I didn't know some short-circiuts. I know it returned a boolean, but I thought it ran through all the array